### ISSUE TITLE
Performance of panning/zooming LineSeries on a high frequency signal

### ISSUE THREAD

**Issue by chriglburri** [oxyplot/oxyplot#1895]
### Description
I noticed a significant difference in the performance when a `LineSignal `of a high frequency sine is plotted vs. a `LineSignal `of a low frequency sine. (It doesn't matter if it is a sine or any other signal. I just used a sine.)

When I use low frequency (where you clearly see the wave of the sine) rendering, panning and zooming are very fast. However when I use high frequency (where you can not identify a curve, the whole plot area is covered by the signal) rendering, panning and zooming are significantly slow.

Apparently the amount of points is not the root of the problem, since I use 100_000 points for each scenario. It looks like the area which is covered by color is crucial. The more of the area is covered the longer the reaction is.

Using the `ScreenPoint Decimator` helps a bit but the difference is still huge.

I was not able to test the behavior with `OxyPlot.SkiaSharp`, since mouse operations seems to not work there. 

### Steps to reproduce

1. Checkout the github project on [https://github.com/chriglburri/OxyPlotRenderingPerformanceDemo](https://github.com/chriglburri/OxyPlotRenderingPerformanceDemo)
2. Read the README.MD how to proceed
3. As written in the README, comment the lines where a signal with high or low frequency is rendered to compare the performance

Platform:

- package id="OxyPlot.Wpf" version="2.1.0" targetFramework="net47"
- .NET version: 4.7.0
- Win10
- Intel i7 3GHz 4 Core
- 32GB RAM
- Intel Iris Xe Graphics

### Expected behavior

High frequency and low frequency signals have a similar performance.

### Actual behavior

High frequency signals react much slower to user interaction than low frequency signals.

**Comment by chriglburri** [comment]
I enhanced my application by another example: the ScopeDataDemo. This is a very common situation for our clients. It renders 4096 points from a csv File. 

You see that the performance on Panning/Zooming is bad when the application runs in full screen. when you resize the window and make it smaller the performance is better even though the same amount of points is shown.

**Comment by chriglburri** [comment]
Apparently the performance is better when we change the version back to 2.0.0. It looks like the problem was introduced in 2.1.0

**Comment by VisualMelon** [comment]
Thanks for your detailed report on this. I'm afraid I don't really have time to investigate these things these days, but if it is a regression in 2.1 then it may be something quite simple. Comparing a profile for 2.1 and 2.0 might make it obvious which method(s) are taking longer and help narrow down what the problem is. Note that high frequency will mean more area covered, but also more data-points after decimation (e.g. line series won't plot data-points that are less than 2 screen points apart, IIRC)

**Comment by chriglburri** [comment]
Thanks @VisualMelon for your response.
I just learned how to inherit the PlotView and overwrite the rendering context to make use of the XamlRenderingContext. 
**This improved the performance significantly**. 
Those who are interested can checkout how I did this in the the repo https://github.com/chriglburri/OxyPlotRenderingPerformanceDemo.

If my company accepts this solution, I will close the Issue.
